woocommerce/woocommerce
# Add license key support to MaxMind GeoLite2 database downloads

### All Submissions:


### Changes proposed in this Pull Request:

Due to the CCPA, MaxMind has removed public access to their GeoLite databases. In order to download the database you must now create an account and generate a license key. A side-effect of this change however is that the GeoLite2 database download in WooCommerce is non-functional. This PR adds support for a license key, as well as lays the groundwork for the MaxMind integration to be moved into a feature plugin.

Closes #25361 and #25370.

### How to test the changes in this Pull Request:

1. Make sure there is no "GeoLite2-Country.mmdb" file in your uploads directory.
2. In your store settings, change the "Default customer location" to one of the Geolocate options. (If you already have been set to one, change to something else and then to a geolocate option.)
3. Without the patch, note that there is no database in your uploads directory. If you check your logs, there will be a geolocation error indicating that it was unable to download the database.
4. Check out the branch and visit the Settings > Integration section. After creating an account on MaxMind's website and generating a key, place that key into the setting and save.
5. Change the default customer location again.
6. Note the archive in the uploads folder.

### Other information:


### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
